crates/ere-cli/src/main.rs

```diff
@@ -0,0 +1,213 @@
+use anyhow::{Context, Error};
+use clap::{Parser, Subcommand};
+use std::{fs, path::PathBuf};
+use zkvm_interface::{Compiler, ProverResourceType, zkVM};
+
+mod serde;
+
+// Compile-time check to ensure exactly one backend feature is enabled
+const _: () = {
+    assert!(
+        (cfg!(feature = "jolt") as u8
+            + cfg!(feature = "nexus") as u8
+            + cfg!(feature = "openvm") as u8
+            + cfg!(feature = "pico") as u8
+            + cfg!(feature = "risc0") as u8
+            + cfg!(feature = "sp1") as u8
+            + cfg!(feature = "zisk") as u8)
+            == 1,
+        "Exactly one zkVM backend feature must be enabled"
+    );
+};
```

Comment:
Re https://github.com/eth-act/ere/pull/66#discussion_r2242879799 it works and this is shorter so I applied it.